### PR TITLE
Update firebase to v10.6.0

### PR DESCRIPTION
**Pull request by yurabakhtin**
https://github.com/humhub/humhub-internal/issues/389

### CODE REVIEW

File: models/ConfigureForm.php [hunk 1]
```diff
@@ -89,6 +93,7 @@ public function rules()
         return [
             [['enableEmailGoService', 'disableAuthChoicesIos'], 'boolean'],
             [['senderId'], 'number'],
+            [['firebaseProjectId', 'firebaseApiKey', 'firebaseAppId', 'firebaseVapidKey'], 'string'],
```
Comment (ArchBlood) [inline models/ConfigureForm.php:96]: Should be set to required.

Comment (yurabakhtin) [reply models/ConfigureForm.php:96]: @ArchBlood The fields "Sender ID" and "Service Account (JSON file)" are not required.
They were used to check the dirver `humhub\modules\fcmPush\driver\Fcm` is configured in previous version.
Now to check the driver is configured we check the 2 old fields and 4 new fields, so I am not sure we should make the 4 new fields required.

File: models/ConfigureForm.php [hunk 2]
```diff
@@ -207,6 +221,10 @@ public function saveSettings()
 
         $module->settings->set('enableEmailGoService', $this->enableEmailGoService);
         $module->settings->set('senderId', $this->senderId);
+        $module->settings->set('firebaseProjectId', $this->firebaseProjectId);
```
Comment (luke-) [inline models/ConfigureForm.php:224]: @yurabakhtin Are you sure we need this information? The JSON file already contains the project ID and keys, for example, doesn't it?
e.g.

```json
{
  "type": "service_account",
  "project_id": "humhub-community",
  "private_key_id": "7acacacacacacacacacacacacacacacacacacacacacacacacacac",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIXXXXXXXXXXXXXXXXXXXXXXXXXx/q\n-----END PRIVATE KEY-----\n",
  "client_email": "firebase-adminsdk-6ynpq@humhub-community.iam.gserviceaccount.com",
  "client_id": "12343262336346346346346346346",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-6ynpq%40humhub-community.iam.gserviceaccount.com",
  "universe_domain": "googleapis.com"
}
```

**Review by luke- — CHANGES_REQUESTED** [review]
@yurabakhtin

File: models/ConfigureForm.php (hunk 2 above)
Comment (yurabakhtin) [reply models/ConfigureForm.php:224]: @luke- If I try to initialize the app without `projectId` then I get the error:

`Could not initialize module: humhub.modules.firebase FirebaseError: Installations: Missing App configuration value: "projectId" (installations/missing-app-config-values).`

or do you mean we should extract the `project_id` from the provided JSON file?

Comment (luke-) [reply models/ConfigureForm.php:224]: Yes, maybe I would be good to extract it.

Comment (yurabakhtin) [reply models/ConfigureForm.php]: @luke- Commit https://github.com/humhub/fcm-push/pull/51/commits/56b3f926fdb71f8902600c26740eed1d641a63a6

File: Events.php [hunk 3]
```diff
@@ -63,22 +63,18 @@ public static function onServiceWorkerControllerInit($event): void
...
+
+            firebase.initializeApp({
+                messagingSenderId: "{$pushDriver->getSenderId()}",
+                projectId: "{$module->getConfigureForm()->getJsonParam('project_id')}",
+                appId: "{$module->getConfigureForm()->firebaseAppId}",
+                apiKey: "{$module->getConfigureForm()->firebaseApiKey}",
             });
+
+            // Initialize Firebase Cloud Messaging and get a reference to the service
+            const messaging = firebase.messaging();
```
Comment (yurabakhtin) [inline Events.php:77]: @luke- Why do we need `const messaging = ` here, if it is not used below?
Maybe only `firebase.messaging();` is enough?

Comment (luke-) [reply Events.php:77]: @yurabakhtin Yes, that should be enough. I've copied it from the official doc.
https://firebase.google.com/docs/cloud-messaging/js/client

I'll remove the const

**Pull request merged**